### ISSUE TITLE
Iteration count not available as a command line argument when looping

### ISSUE THREAD

**Issue by trobob** [spine-tools/Spine-Toolbox#2105]
**Preamble:**

I was recently trying to use the looping feature in Spine Toolbox. This can be done by holding alt then clicking from one item to the next in the workflow. Then when selecting the blue arrow between the items, there is a place to add a tool specification or python script containing the code for the loop. It is also possible to add command line arguments from the available resources. So far, so good.

**The issue:**

I would like to use the iteration count as a variable for selecting data in a tool the loop is connected to at the arrow tip. However, the iteration count is not currently forwarded as an argument/variable via the looping tool. See image below.

![spine toolbox loop](https://user-images.githubusercontent.com/122376655/235480931-d5fcd3ff-9640-4573-9fcf-18b7dddfe1ee.jpg)

I am currently using a workaround that writes the current iteration count to an intermediate "dummy database", then the iteration count is read from the database by the next tool in the workflow. This not ideal and takes extra effort.

**Request:**

Can the iteration count in the loop autmatically be made available as a variable to any connected tool? Or at least be made available as a command line argument that can be passed to the next tool manually in the same manner as the datastores?

**Comment by soininen** [comment]
A bit different issue, but it would be nice to have the count available as a command line argument in Jump arrow's properties tab, too. Currently, it is passed to the loop condition script when there are no other arguments and it is not really documented anywhere. I think we should make it more explicit.

**Comment by manuelma** [comment]
I agree completely @soininen we should model it as a ProjectItemResource of sorts and make it visible.

**Comment by jkiviluo** [comment]
As @trobob has a workaround, I'm putting this to 1.0. Can be moved if needed.
